### ISSUE TITLE
Bad message and not working workaround

### ISSUE THREAD

**Issue by remicollet** [jcupitt/php-vips#27]
```
$ php -v
php-vips-ext: unable to lock libvips -- graceful may be unreliable
PHP 7.0.14 (cli) (built: Dec  7 2016 09:36:47) ( NTS )
```

As i understand, this message is related to #26 so perhaps this message should be display only for "mod_php", not for other SAPI ?

**Comment by jcupitt** [comment]
Oh good point, thanks @remicollet. 

I'll have a look. I don't suppose you know of an easy way to spot if you're running inside apache?

**Comment by remicollet** [comment]
More, `if (!dlopen("libvips.so", RTLD_LAZY | RTLD_NODELETE)) {` can't work, libvips.so is usually not installed (part of "devel" stuff), only the versioned library needed at runtime (e.g. libvips.so.42.6.4)

**Comment by remicollet** [comment]
More, the printf call is absolutely not acceptable, should use the SAPI error logger.

**Comment by remicollet** [comment]
> I don't suppose you know of an easy way to spot if you're running inside apache?

You can test `strcmp(sapi_module.name, "apache2handler"))`

**Comment by jcupitt** [comment]
Hello again, thank you for taking the time to look at this @remicollet.

Does this look better?

https://github.com/jcupitt/php-vips-ext/commit/ed9f4243a34c0f57c8c57485e2650bbaa4e2a38e#diff-b1cc89f7ef630413232cfad4ba9bdbf2R1625

It now tests for apache, it uses the SAPI logger, and it uses the API number in the .soname. I updated the `notes` section in `package.xml` as well.

I tested under apache and it seems to survive `graceful` successfully, and I tried at the CLI and it runs without strange console output. All the tests pass too.
